src/generic/problem.h

```diff
@@ -3100,35 +3103,69 @@ namespace oomph
   //=======================================================================
   /// A class to handle errors in the Newton solver
   //=======================================================================
-  class NewtonSolverError
+  class NewtonSolverError : public OomphLibError
   {
-  public:
+  private:
```

Comment:
If we make the public variables private, then we are changing the interface of the class. I've added getter functions but if someone else's code tries to access the class's variable then it'll now throw an error.

Reply:
If they read public data they deserve everything they get. Just make sure there's an explicit comment in the commit log.

Reply:
I disagree with this broad statement that public data should not be read. The whole point of error objects is that they are essentially structures with a number of diagnostic variables that should, in my view, be public rather than going through the overhead of set and get functions. Set should never be needed because the only time variables are set is during creation. The object is created, thrown and destroyed after being caught or rethrown, so there is no potential for unexpected side effects. No benefit to data encapsulation in my view.